dan_layer/engine/src/runtime/impl.rs

```diff
@@ -832,6 +879,14 @@ impl<TTemplateProvider: TemplateProvider<Template = LoadedTemplate>> RuntimeInte
 
                 self.tracker.write_with(|state| {
                     let vault_lock = state.lock_substate(&SubstateId::Vault(vault_id), LockFlag::Write)?;
+                    let bucket = state.take_bucket(bucket_id)?;
+
+                    // Emit a builtin event for the deposit
+                    let amount = match bucket.resource_type() {
+                        ResourceType::Fungible => Some(bucket.amount()),
+                        _ => None,
+                    };
```

Comment:
All resources support amount 

```suggestion
                    let amount = bucket.amount();
```